src/components/PlayerComponent.react.js

```diff
@@ -0,0 +1,214 @@
+import React, {Component} from 'react';
+import ReactPlayer from 'react-player';
+import PropTypes from 'prop-types';
+
+/**
+ * A Dash component for playing a variety of URLs, including file paths,
+ * YouTube, Facebook, Twitch, SoundCloud, Streamable, Vimeo, Wistia, Mixcloud,
+ * and DailyMotion.
+ */
+export default class PlayerComponent extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {value: props.value};
+
+        this.updateCurrentTime = this.updateCurrentTime.bind(this);
+        this.updateInstanceMethods = this.updateInstanceMethods.bind(this);
+        this.setSeekTo = this.setSeekTo.bind(this);
+        setInterval(this.updateCurrentTime, 40);  // 25 FPS
+        setInterval(this.updateInstanceMethods, 500);
+    }
+
+    updateCurrentTime(){
+        const {setProps} = this.props;
+        const currentTime = this.refs.player.getCurrentTime();
+        if (setProps  !== null) {
+            setProps({currentTime: currentTime});
+        }
+    }
+
+    updateInstanceMethods(){
+        const {setProps} = this.props;
+        const secondsLoaded = this.refs.player.getSecondsLoaded();
+        const duration = this.refs.player.getDuration();
+
+        if (setProps  !== null) {
+            setProps({
+                secondsLoaded: secondsLoaded,
+                duration: duration
+            });
+        }
+    }
+
+    /**
+     * Applies the seekTo() method to the player if the seekTo prop
+     * contains any value, then set seekTo to null. If seekTo was already
+     * null, it doesn't do anything.
+     */
+    setSeekTo(){
+        const {
+            seekTo,
+            setProps
+        } = this.props;
+
+        if (seekTo !== null && setProps !== null){
+            this.refs.player.seekTo(seekTo);
+            setProps({seekTo: null});
```

Comment:
This is very clever. I can't actually think of any other way to do this. It's a little bit of a weird pattern because `seekTo` ends up meaning two things: a value needs to be seeked to, or a seek _previously happened_. It's also a bit of an anti-pattern because `setProps` can be expensive: it currently re-renders the entire dash app (although it shouldn't) and in the future it will just rerender the component.

Usually, we'd do this type of pattern by comparing `this.props.seekTo` with `prevProps.seekTo` or `nextProps.seekTo` and if they are different, then we'd actually call the method:
```
if (nextProps.seekTo !== this.props.seekTo) {
  this.refs.player.seekTo(nextProps.seekTo);
}
```
However, we can't do that here because the user may set the _same_ `seekTo` value multiple times in a row. e.g. a button that sets `seekTo` to 40. 

It's odd because if this was an object, then our equality check would work even if the prop was updated even if the object's values were the same. That is:
```
props.obj = {'seekTo: 3}

// then a callback happens, and dash updates the props:
newProps = R.merge(props, {obj: {seekTo: 3}}
newProps.obj === props.obj // false: the obj is a new instance
newProps.obj.seekTo === props.obj.seekTo // true, the number is the same
```

I suppose another way we could get around it is by making the seekTo property some kind of object like:
```
seekTo: {value: 40, key: some_unique_id}
```
and ask that our users provide a unique ID when they want to do a seekTo update. That way, we could do an equality check to see if this was a new property or an old property. That's also pretty unconventional.

So, let's keep this as is. This might just be something that we need to improve on in the dash-renderer side: provide more information to the components about _which_ property has changed.

Reply:
That's an interesting issue. At the moment, having `nextProps.seekTo !== this.props.seekTo` not returning the same thing when they have the same value will likely cause an infinite loop, because every time `setSeekTo()` is called, the component gets updated, which triggers `componentDidUpdate()`; it in return calls `setSeekTo()`, etc.

Will be waiting for dash-renderer to update, so that we can also update currentTime and updateInstanceMethods to be non-interval dependent.

Reply:
> Will be waiting for dash-renderer to update, so that we can also update currentTime and updateInstanceMethods to be non-interval dependent.

Just to be clear, these changes won't get into `dash-renderer` for a long time likely